android/java/org/chromium/chrome/browser/crypto_wallet/observers/KeyringServiceObserverImpl.java

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.observers;
+
+import org.chromium.mojo.system.MojoException;
+
+public class KeyringServiceObserverImpl
+        implements org.chromium.brave_wallet.mojom.KeyringServiceObserver {
+    public interface KeyringServiceObserverImplDelegate {
```

Comment:
not sure I understand what do you mean by a listener. The observer is the listener and we have a class that implements the observer and it delegates some methods executions to diff instances. Could you make an example of what kind of listener do you have in mind and what are advantages of it over the current implementation?

Reply:
An example of using delegates in chromium's code base https://source.chromium.org/chromium/chromium/src/+/main:components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogController.java;l=73?q=PermissionDialogController.java&ss=chromium%2Fchromium%2Fsrc

Reply:
[PermissionDialogDelegate](https://source.chromium.org/chromium/chromium/src/+/main:components/permissions/android/java/src/org/chromium/components/permissions/PermissionDialogDelegate.java;drc=91e23cd891fa63eca6d32c16a30a977a87c9f587;bpv=1;bpt=1;l=21?gsn=PermissionDialogDelegate&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Djava%3Fpath%3Dorg.chromium.components.permissions.PermissionDialogDelegate%23340d565bc3b938cf0f1595e66618dbd62f2ac6e26ec0a9a74b388702a62cbdd5) is handling the communication between native layer and java code.
Additionally, we should follow one convention for similar types of implementation to avoid confusion.

Reply:
here is an example without native https://source.chromium.org/chromium/chromium/src/+/main:android_webview/java/src/org/chromium/android_webview/AwContents.java;l=239;drc=91e23cd891fa63eca6d32c16a30a977a87c9f587;bpv=1;bpt=1?q=AwContents.java&ss=chromium%2Fchromium%2Fsrc.
And yes we follow the same convention everywhere. The convention is how chromium code base does it as we work with it.

Reply:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/autofill/CardUnmaskPrompt.java;l=89;drc=91e23cd891fa63eca6d32c16a30a977a87c9f587?q=CardUnmaskPrompt.java
https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/compositor/overlays/strip/StripLayoutTab.java;l=58;drc=91e23cd891fa63eca6d32c16a30a977a87c9f587?q=overlays%2Fstrip%2FStripLayoutTab.java
a few more examples of what is Delegate and how it is used ^

Reply:
here is an example of Listener https://source.chromium.org/chromium/chromium/src/+/main:android_webview/java/src/org/chromium/android_webview/WebMessageListenerHolder.java
so both can be used interchangeably.

I personally prefer to use android framework conventions with Listeners (Like [GestureListner](https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/view/GestureDetector.java#64)).

Reply:
So what do we do? Do we correct each others PRs when review to what we prefer? I will correct yours to Delegate you mine to Listener? The review is not about an ask to change names to what you see them and what you like them to look better for you. The review is more about checking a logic. There are several ways almost in any task that it can be done. No point to impose your way of how should it look like unless the implementer made it obviously wrong. And no point to spend time on such discussions in the future.

Reply:
We have been using listeners at many places in brave for data passing, and since we have more control over our codebase then we should prefer to follow a single convention as it makes it easy for us and new devs to follow and search through the code-base, rest is upto you whether you would like to follow conventions or not. The PR is has been already approved so feel free to proceed.